app/uk/gov/hmrc/exports/services/NotificationService.scala

```diff
@@ -39,15 +39,23 @@ class NotificationService @Inject()(submissionRepository: SubmissionRepository,
   private val databaseDuplicateKeyErrorCode = 11000
 
   def getNotifications(submission: Submission): Future[Seq[Notification]] = {
-    val conversationIds = submission.actions.map(_.id)
-    notificationRepository.findNotificationsByActionIds(conversationIds).map { notifications =>
-      notifications.map { notification =>
-        notification.copy(errors = notification.errors.map { error =>
-          val url = error.pointer.flatMap(WCOPointerMappingService.getUrlBasedOnErrorPointer)
-          error.addUrl(url)
-        })
-      }
+
+    def updateErrorUrls(notification: Notification) = {
+      val updatedDetails = notification.details.map(
+        details =>
+          details.copy(errors = details.errors.map { error =>
+            val url = error.pointer.flatMap(WCOPointerMappingService.getUrlBasedOnErrorPointer)
+            error.addUrl(url)
+          })
+      )
+
+      notification.copy(details = updatedDetails)
     }
+
+    val conversationIds = submission.actions.map(_.id)
+    notificationRepository
+      .findNotificationsByActionIds(conversationIds)
+      .map(_.map(updateErrorUrls(_)))
```

Comment:
should we use `flatMap` here?

Reply:
First map is on a Future, second one is on a Seq.